src/synapse/api.py

```diff
@@ -182,15 +192,26 @@ def get_version() -> str:
         session.mount("http://", HTTPAdapter(max_retries=retries))
         res = session.get(VERSION_URL, timeout=10)
         res.raise_for_status()
-        server_version = res.json()["server_version"]
-        version_match = re.search(r"^([^\s(]+)", server_version)
-        if version_match:
-            return version_match.group(1)
-        return server_version
-    except (
-        requests.exceptions.ConnectionError,
-        requests.exceptions.Timeout,
-        requests.exceptions.HTTPError,
-    ) as exc:
-        logger.exception("Failed to request %s : %r", VERSION_URL, exc)
-        raise NetworkError(f"Failed to request {VERSION_URL}.") from exc
+        res_json = res.json()
+        server_version = res_json.get("server_version", None)
+        if server_version is None:
+            # Exception not in docstring because is captured.
+            raise VersionNotFoundError(  # noqa: DCO053
+                f"There is no server_version in JSON output: {res_json}"
+            )
+        version_match = re.search(SYNAPSE_VERSION_REGEX, server_version)
+        if not version_match:
+            # Exception not in docstring because is captured.
```

Comment:
Also not following this?

Reply:
Closing this one in favor of the first answer.